Packs/DigitalGuardian/Integrations/DigitalGuardianARCEventCollector/DigitalGuardianARCEventCollector.py

```diff
@@ -22,239 +24,246 @@ class Client(BaseClient):
+    demisto.debug('Fetching events')
+    raw_response = client.export_events()
+
+    demisto.info(f"Pulled {raw_response['total_hits']} events from {VENDOR}")
+    events = create_events_for_push(raw_response, limit)
+
+    last_run = {key: value for key, value in raw_response.items() if key in ('bookmark_values', 'search_after_values')}
 
+    return events, last_run
 
-def create_events_for_push(event_list: list, last_time: str, id_list: list, limit: int) -> Tuple[list, str, list]:
+
+def create_events_for_push(raw_response: dict, limit: int | None = None) -> EventsGenerator:
     """
-       Create events for pushing them and prepares the values for next_run save
-       Args:
-           event_list (list): list of events
-           last_time (str): time of last event from previous run
-           id_list (list): list of id's ingested
-           limit (int): max_fetch
-       Returns:
-           list: list of events
-           last_time: updates time of last event
-           id_list: list of id's
+    Yields key-value dictionaries of distinct events from the raw API response and adds the _time key to the events.
+    Args:
+        raw_response (dict): Export profile events raw API response.
+        limit (int | None): Optional value to limit the number of yielded events.
+    Yields:
+        dict: Event from raw response.
     """
+    event_fields = [field['name'] for field in raw_response['fields']]
+    events_data = raw_response['data']
+
+    ids = set()
     index = 0
-    event_list_for_push = []
-    demisto.debug('Checking duplications and creating events for pushing to XSIAM')
-    for event in event_list:
-        inc_time = event.get("inc_mtime")
-        if last_time:
-            last_time_date = arg_to_datetime(arg=last_time, required=True).date()   # type: ignore[union-attr]
-            event_date = arg_to_datetime(arg=inc_time, required=True).date() if inc_time else None   # type: ignore[union-attr]
-            if (inc_time and inc_time < last_time) or event.get("dg_guid") in id_list:
-                continue
-            if last_time_date == event_date and event.get("dg_guid"):
-                id_list.append(event.get("dg_guid"))
-            elif event.get("dg_guid"):
-                id_list = [event.get("dg_guid")]
-        else:
-            if event.get("dg_guid"):
-                id_list.append(event.get("dg_guid"))
-        event_list_for_push.append(event)
-        last_time = inc_time if inc_time else last_time
-        index += 1
-        if index == limit:
+
+    for event_data in events_data:
+        if limit is not None and index == limit:  # No API-side limit param (needs to be managed on our end)
             break
-    return event_list_for_push, last_time, id_list
+
+        event = dict(zip(event_fields, event_data, strict=True))
+        event_id = event['dg_guid']
+
+        if event_id in ids:
+            continue  # Skip duplicate event
```

Comment:
let's leave some breadcrumbs behind - create a `collections.Counter` of the ids, and log it later as a dictionary (preferably sorted desc)